### PR TITLE
Fixed remote pin warning when in offline mode

### PR DESCRIPTION
**Pull request by Zanda256**
Added function to check if node is in offline mode and returns a bool.
If bool is true, browse the local data store, if CID is present in local datastore, a warning is issued.

### CODE REVIEW

**Comment by BigLep** [comment]
We'll take a look at this during weekly triage on Friday if not sooner.

File: core/commands/pin/remotepin.go [hunk 1]
```diff
@@ -173,6 +175,34 @@ NOTE: a comma-separated notation is supported in CLI for convenience:
 		if err != nil {
 			return err
 		}
+		//ifconfig.co
+		checkConnection := func() bool {
+			_, err := http.Get("ifconfig.co")
+			return err == nil
+		}
```
Comment (lidel) [inline core/commands/pin/remotepin.go:182]: `checkConnection` needs to be removed.

We can't use a third party DNS name + HTTP service for doing this test.
If it goes offline or can't be reached, our _online_ users will get a false-negative.

Checking `node.IsOnline` is the best we can do without introducing centralized services.

File: core/commands/pin/remotepin.go [hunk 2]
```diff
@@ -173,6 +175,34 @@ NOTE: a comma-separated notation is supported in CLI for convenience:
...
+		offline, _ := req.Options["offline"].(bool)
+
+		if offline || !node.IsOnline || !netConnection {
+			fmt.Println("checking connection")
+			allKeys, err := node.Blockstore.AllKeysChan(ctx)
+			if err != nil {
+				return err
+			}
+			for k := range allKeys {
+				if k == rp.Cid() {
```
Comment (lidel) [inline core/commands/pin/remotepin.go:194]: This is an equivalent of doing `ipfs refs local | grep CID`, which is bit rough :)

There are better ways of checking if a block is in the datastore, for example you could use `ipfs dag stat --offline`:

```
$ ipfs dag stat /ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR --offline
Size: 119776, NumBlocks: 1

$ ipfs dag stat /ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMns --offline
Error: merkledag: not found
```

Mind replacing this for loop with something like:

```
	_, err := api.Dag().Get(req.Context, rp.Cid())
	if err != nil {
        # getting remote block in offline mode will produce Error: merkledag: not found
		return err
	}
```

**Review by lidel — CHANGES_REQUESTED** [review]
Thank you @Zanda256!
We need to change two things  before this can be merged, see comments inline.

File: core/commands/pin/remotepin.go (hunk 2 above)
Comment (Zanda256) [reply core/commands/pin/remotepin.go:194]: Thank you for the help @lidel , I will make the changes and re-submitt the pr.

**Comment by lidel** [comment]
Thank you for poking at this!
Superseded by https://github.com/ipfs/go-ipfs/pull/8621

**Pull request closed without merging**